typescript/sdk/src/core/EvmCoreModule.ts

```diff
@@ -0,0 +1,267 @@
+import { Mailbox } from '@hyperlane-xyz/core';
+import { Address, ProtocolType, rootLogger } from '@hyperlane-xyz/utils';
+
+import {
+  attachContractsMap,
+  serializeContractsMap,
+} from '../contracts/contracts.js';
+import { HyperlaneAddresses } from '../contracts/types.js';
+import { CoreConfig } from '../core/types.js';
+import { HyperlaneProxyFactoryDeployer } from '../deploy/HyperlaneProxyFactoryDeployer.js';
+import {
+  ProxyFactoryFactories,
+  proxyFactoryFactories,
+} from '../deploy/contracts.js';
+import { HyperlaneIsmFactory } from '../ism/HyperlaneIsmFactory.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { EthersV5Transaction } from '../providers/ProviderType.js';
+import { ChainNameOrId } from '../types.js';
+
+import {
+  HyperlaneModule,
+  HyperlaneModuleArgs,
+} from './AbstractHyperlaneModule.js';
+import { EvmIcaModule } from './EvmIcaModule.js';
+import { HyperlaneCoreDeployer } from './HyperlaneCoreDeployer.js';
+import { CoreFactories } from './contracts.js';
+
+// Partial CoreConfig because will be filled once .deploy() is called
+type DeployedAdresses = Partial<HyperlaneAddresses<CoreFactories>> & {
+  testRecipient?: Address;
+  timelockController?: Address;
+  interchainAccountRouter?: Address;
+  interchainAccountIsm?: Address;
+  deployedIsmFactoryFactories: HyperlaneAddresses<ProxyFactoryFactories>;
+};
+
+export class EvmCoreModule extends HyperlaneModule<
+  ProtocolType.Ethereum,
+  CoreConfig,
+  DeployedAdresses
+> {
+  public readonly chainName: string;
+  protected logger = rootLogger.child({ module: 'EvmCoreModule' });
+  protected hyperlaneCoreDeployer: HyperlaneCoreDeployer;
+
+  protected constructor(
+    protected readonly multiProvider: MultiProvider,
+    args: HyperlaneModuleArgs<CoreConfig, DeployedAdresses>,
+  ) {
+    super(args);
+    this.chainName = this.multiProvider.getChainName(this.args.chain);
+
+    // Deploy IsmFactory to be used in CoreDeployer
+    const ismFactory = new HyperlaneIsmFactory(
+      attachContractsMap(
+        { [this.chainName]: this.args.addresses.deployedIsmFactoryFactories },
+        proxyFactoryFactories,
+      ),
+      multiProvider,
+    );
+
+    // Initalize Deployer
+    this.hyperlaneCoreDeployer = new HyperlaneCoreDeployer(
+      multiProvider,
+      ismFactory,
+    );
+  }
```

Comment:
iiuc this means the module will be affected by the deployer's caching
is that intentional?

Reply:
Removed `this.hyperlaneCoreDeployer`. Deployer is used only in static deploy now